### PR TITLE
[FLINK-2445] Add tests for HadoopOutputFormats

### PR DESCRIPTION
**Pull request by mliesenberg**
as discussed in [PR 1486](https://github.com/apache/flink/pull/1486) a new PR addressing the comments.

### CODE REVIEW

**Comment by mliesenberg** [comment]
build failure because of
- [FLINK-1445](https://issues.apache.org/jira/browse/FLINK-1455) in [build job](https://travis-ci.org/apache/flink/jobs/114046004)
- [FLINK-3530](https://issues.apache.org/jira/browse/FLINK-3530) see [log](https://s3.amazonaws.com/archive.travis-ci.org/jobs/114046006/log.txt) in [build job](https://travis-ci.org/apache/flink/jobs/114046006)

[build job](https://travis-ci.org/apache/flink/jobs/114046008) fails because one of the newly added tests failed, will check and update the PR.

**Comment by mliesenberg** [comment]
I could not reproduce the issue locally, I rebased the PR onto the current master and pushed that again.

Thankful for any pointers, the issue was the following: `testOpen` and `testFinalizeGlobal` were not passing because of `RuntimeExceptions`. `TaskAttemptContext` and `JobContext` could not be created.

**Comment by mliesenberg** [comment]
it's only the build job with with the `hadoop.profile=1` option. for the others the tests are passing.

File: flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatTest.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+import static org.mockito.Mockito.*;
+
+public class HadoopOutputFormatTest {
+
+    @Test
+    public void testOpen() throws Exception {
+
+        OutputFormat<String, Long> dummyOutputFormat = mock(DummyOutputFormat.class);
+        DummyOutputCommitter outputCommitter = mock(DummyOutputCommitter.class);
+        JobConf jobConf = mock(JobConf.class);
```
Comment (fhueske) [inline flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatTest.java]: This line needs to be replaced by `JobConf jobConf = spy(new JobConf());`.

The Hadoop code copies the `JobConf` and access additional fields that cannot be mocked.

File: flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatTest.java [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        outputFormat.writeRecord(new Tuple2<>("key", 1L));
+
+        verify(recordWriter, times(1)).write(anyString(), anyLong());
+    }
+
+    @Test
+    public void testFinalizeGlobal() throws Exception {
+        OutputFormat<String, Long> dummyOutputFormat = mock(DummyOutputFormat.class);
+        DummyOutputCommitter outputCommitter = mock(DummyOutputCommitter.class);
+        JobConf jobConf = mock(JobConf.class);
```
Comment (fhueske) [inline flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatTest.java]: This line needs to be replaced by `JobConf jobConf = spy(new JobConf());`.

The Hadoop code copies the `JobConf` and access additional fields that cannot be mocked.

File: flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatTest.java [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+        when(jobConf.getOutputCommitter()).thenReturn(outputCommitter);
+
+        HadoopOutputFormat<String, Long> outputFormat = new HadoopOutputFormat<>(dummyOutputFormat, jobConf);
+
+        outputFormat.open(1, 1);
+
+        verify(jobConf, times(1)).getOutputCommitter();
+        verify(outputCommitter, times(1)).setupJob(any(JobContext.class));
```
Comment (fhueske) [inline flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatTest.java]: change `times(1)` to `times(2)`

File: flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatTest.java [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.java.hadoop.mapred;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.hadoop.conf.Configurable;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.mapred.*;
```
Comment (fhueske) [inline flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatTest.java]: No star imports please

File: flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapreduce/HadoopOutputFormatTest.java [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.java.hadoop.mapreduce;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.hadoop.conf.Configurable;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.*;
```
Comment (fhueske) [inline flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapreduce/HadoopOutputFormatTest.java]: No star imports please

File: flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapreduce/HadoopOutputFormatTest.java [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
+
+
+public class HadoopOutputFormatTest {
```
Comment (fhueske) [inline flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapreduce/HadoopOutputFormatTest.java]: This file uses mixed tabs and spaces for indention. Please use tabs only.

File: flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatTest.java [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+import org.apache.hadoop.util.Progressable;
+import org.junit.Test;
+import org.mockito.Matchers;
+
+import java.io.IOException;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.*;
+
+public class HadoopOutputFormatTest {
```
Comment (fhueske) [inline flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatTest.java]: Please change the indention from spaces to tabs.

**Comment by fhueske** [comment]
Thanks for the PR!
I found the issue for the Hadoop 1 failures (see inline comments).
There are also a few style issues (star imports, space indention).

File: flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatTest.java (hunk 1 above)
Comment (mliesenberg) [inline flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatTest.java]: I suppose that applies to the `JobConf` in `testFinalizeGlobal` as well?
Is there a way for me to test that locally?

Comment (fhueske) [inline flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatTest.java]: Yes, I did `mvn -Dhadoop.profile=1 clean install` on my local machine.

Comment (mliesenberg) [inline flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatTest.java]: oh well. that's easy then. will update the PR soon then.

**Comment by fhueske** [comment]
Thanks for the quick update. Will merge this PR

**Comment by fhueske** [comment]
Hi @mliesenberg, I merged this PR but it wasn't correctly closed. Can you do it? Thanks

**Pull request closed without merging**